sql/builder/builder.go

```diff
@@ -87,52 +97,87 @@ func FilterFrom(operations Operations) string {
 			queryBuilder.WriteString(" and")
 		}
 
-		if op.Token == In {
-			values, ok := op.Value.([][]byte)
-			if !ok {
-				panic("value for 'In' token must be a slice of []byte")
+		if len(op.Group) > 0 {
+			queryBuilder.WriteString(" (")
+			for k, groupOp := range op.Group {
+				if k != 0 {
+					queryBuilder.WriteString(fmt.Sprintf(" %s", op.GroupOperator))
+				}
+				if groupOp.Token == In {
+					values, ok := groupOp.Value.([][]byte)
+					if !ok {
+						panic("value for 'In' token must be a slice of []byte")
+					}
+					params := make([]string, len(values))
+					for j := range values {
+						params[j] = fmt.Sprintf("?%d", bindIndex)
+						bindIndex++
+					}
+					fmt.Fprintf(&queryBuilder, " %s%s %s (%s)", groupOp.Prefix, groupOp.Field, groupOp.Token,
+						strings.Join(params, ", "))
+				} else {
+					fmt.Fprintf(&queryBuilder, " %s%s %s ?%d", groupOp.Prefix, groupOp.Field, groupOp.Token, bindIndex)
+					bindIndex++
+				}
 			}
-			params := make([]string, len(values))
-			for j := range values {
-				params[j] = fmt.Sprintf("?%d", bindIndex)
+			queryBuilder.WriteString(" )")
+		} else {
+			if op.Token == In {
+				values, ok := op.Value.([][]byte)
+				if !ok {
+					panic("value for 'In' token must be a slice of []byte")
+				}
+				params := make([]string, len(values))
+				for j := range values {
+					params[j] = fmt.Sprintf("?%d", bindIndex)
+					bindIndex++
+				}
+				fmt.Fprintf(&queryBuilder, " %s%s %s (%s)", op.Prefix, op.Field, op.Token, strings.Join(params, ", "))
+			} else {
+				fmt.Fprintf(&queryBuilder, " %s%s %s ?%d", op.Prefix, op.Field, op.Token, bindIndex)
 				bindIndex++
 			}
-			fmt.Fprintf(&queryBuilder, " %s%s %s (%s)", op.Prefix, op.Field, op.Token, strings.Join(params, ", "))
-		} else {
-			fmt.Fprintf(&queryBuilder, " %s%s %s ?%d", op.Prefix, op.Field, op.Token, bindIndex)
-			bindIndex++
 		}
 	}
 
 	for _, m := range operations.Modifiers {
 		queryBuilder.WriteString(fmt.Sprintf(" %s %v", string(m.Key), m.Value))
 	}
-
 	return queryBuilder.String()
 }
 
 func BindingsFrom(operations Operations) sql.Encoder {
 	return func(stmt *sql.Statement) {
 		bindIndex := 1
 		for _, op := range operations.Filter {
-			switch value := op.Value.(type) {
-			case int64:
-				stmt.BindInt64(bindIndex, value)
-				bindIndex++
-			case []byte:
-				stmt.BindBytes(bindIndex, value)
-				bindIndex++
-			case types.EpochID:
-				stmt.BindInt64(bindIndex, int64(value))
-				bindIndex++
-			case [][]byte:
-				for _, v := range value {
-					stmt.BindBytes(bindIndex, v)
-					bindIndex++
+			if len(op.Group) > 0 {
+				for _, groupOp := range op.Group {
+					bindValue(stmt, &bindIndex, groupOp.Value)
```

Comment:
Usually in Go, APIs return the updated value instead of using a pointer. Consider:
```suggestion
					bindIndex = bindValue(stmt, bindIndex, groupOp.Value)
```